src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -438,7 +438,22 @@ public virtual bool CanExecuteCode(string text)
 
         public Task<ExecutionResult> ResetAsync(bool initialize = true)
         {
-            GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
+            // This is an implementation of ResetAsync on IInteractiveEvaluator interface.
+            // C# and VB Interactive window should not be calling this method at all
+            // as they now call reset through IInteractiveEvaluator2
+
+            // Now we pass on this call to IInteractiveEvaluation2 implementation
+            // which maintains old behaviour when isFromPrompt was not there
+            
+            return ResetAsync(initialize, true);
+        }
+
+        public Task<ExecutionResult> ResetAsync(bool initialize = true, bool isFromPrompt = true)
+        {
+            if (!isFromPrompt)
+            {
+                GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
```

Comment:
I feel like adding new interfaces and parameterizing this method aren't "worth it" just to display ```#reset``` when you click the reset button in the Interactive Window.

What were the other cases you were mentioning where it seemed like these methods would need to be parameterized?  (sorry, I forget our IM conversation)